controllers/restaurants.controller.js

```diff
@@ -1,34 +1,80 @@
 const db = require('../db/index');
 
-exports.showAllRestaurants = async (req, res, next) => {
-    try {
-        const restaurants = await db.any('SELECT * FROM OwnedRestaurants ORDER BY rname');
-        res.render('restaurants', {
-            title: 'All Restaurants',
-            restaurants: restaurants
-
-        });
-    } catch (e) {
-        next(e);
-    }
+/* show all restaurants or selected restaurants based on req.query */
+exports.showRestaurants = async (req, res, next) => {
+  try {
+    const restaurants = await queryDbFromReqQuery(
+      'SELECT DISTINCT rname FROM OwnedRestaurants NATURAL JOIN HasTimeslots',
```

Comment:
`ORDER BY rname`

Reply:
Problem is that `ORDER BY` should be after `WHERE` clause. Need to find a workaround for this.

Reply:
For future consideration!